.github/workflows/python-ci.yml

```diff
@@ -0,0 +1,101 @@
+name: Python CI
+permissions:
+  contents: read
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'python/**'
+      - '.github/workflows/python-ci.yml'
+  pull_request:
+    paths:
+      - 'python/**'
+      - '.github/workflows/python-ci.yml'
+
+defaults:
+  run:
+    working-directory: ./python
+
+jobs:
+  lint-and-format:
+    name: Lint and Format Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
```

Comment:
FYI: `checkout@v5` is latest.

Reply:
would be nice to upgrade, but will keep for now as I'm not that familiar with GitHub actions